### ISSUE TITLE
Suggestion: Better handling of "&" artists

### ISSUE THREAD

**Issue by Velocidensity** [krateng/maloja#246]
Currently, they're indiscriminately split, unless a user adds a rule saying otherwise. This is non-ideal, since the rules have to be manually maintained, and scrobbles rechecked when they're changed. This could be improved, by querying Last.fm or another data source to see if the artist with an ampersand exists before processing. If yes, they would be saved as one artist, if not, split as it is done now. This would add a single extra request for every new artist, after that this data can be saved. This would slow down the data import process, but shouldn't cause too many issues with individual scrobbles.

**Comment by Velocidensity** [comment]
Here's an example which I can't get to parse correctly, even with custom rules.
`PSY, SUGA – That That (prod. & feat. SUGA of BTS)` (https://open.spotify.com/track/7GNRUsU3M4XNDDB9xle5Dz)

This is currently parsed as:
- Artists: `Psy`, `SUGA of BTS)` 
- Title: `That That (prod. &`

Correct parsing:
- Artists: `Psy`, `SUGA` OR `Psy`, `SUGA of BTS` 
- Title: `That That`

**Comment by FoxxMD** [comment]
To be fair that is an extremely cursed track title wrt parsing

**Comment by SirMartin** [comment]
I found an even funnier case of this, I found that the scrobbled of Munford & Sons, are splitter into Munford and other artists named Sons, were are the same band.

![image](https://github.com/krateng/maloja/assets/854646/b6aa3bb6-7c31-4960-a92b-4af22c37d84d)
![image](https://github.com/krateng/maloja/assets/854646/a530f51a-ed0f-4678-803c-563079d038ab)

Not sure how needs to be fixed, I will try to implement a custom rule for it, and merge them.

**Comment by Velocidensity** [comment]
Yes, artists with "&" always get split, it's quite annoying, and I had to write a custom ruleset for those which appear in my library, but it's quite annoying to do every time, and there is no way to re-parse scrobbles imported from Last.fm, so I have ended up restarting my database with a fresh Last.fm import a couple of times.

**Comment by SirMartin** [comment]
You can merge it, its hard the first time, but after a couple of merges it's quite understandable, but yes, it's annoying. Not sure how the rule should be. If the scrobbler are sending more precise data and Maloja should be able to interpretate in other way. In my case I use Spotify with multiscrobbler, not sure how is the one sending confuse data.

**Comment by duckfromdiscord** [comment]
> You can merge it, its hard the first time, but after a couple of merges it's quite understandable, but yes, it's annoying. Not sure how the rule should be. If the scrobbler are sending more precise data and Maloja should be able to interpretate in other way. In my case I use Spotify with multiscrobbler, not sure how is the one sending confuse data.

@SirMartin Maloja has its own rules as to what it does with artist names, multiscrobbler just sends them.

**Comment by SirMartin** [comment]
Hi I am having issues with this again, I could fix the issue with Mumford & Sons with the rules, but I cannot make it work now.

I have this artist, previously known as Years & Years and now is this **Olly Alexander (Years & Years)**
![image](https://github.com/krateng/maloja/assets/854646/66dd35e0-aace-4073-97b0-f55f4e1eadff)

I added as you can see in the rules file that I used for the others and worked fine for Mumford & Sons but it doesn't for this one.

```
belongtogether  Megan & Liz
belongtogether  Bell, Book & Candle
belongtogether  K/DA
belongtogether  Darth & Vader
belongtogether  Case & Point
belongtogether  Selena Gomez & The Scene
belongtogether  Gerry & The Pacemakers
belongtogether  AC/DC
belongtogether  Au/Ra
belongtogether  Mumford & Sons
belongtogether  Olly Alexander (Years & Years)
```

I copied from the logs from Maloja to have it exactly as multiscrobbler sent to Maloja, but still doesn't work, I restarted the container multiple times, but it still split it.

I tested with the latest master version and with the 2.2.2 image.

Any ideas?
